application/dashboard/trello_service.py

```diff
@@ -97,11 +100,11 @@ def _map_card(self, card):
         return obj
 
     @staticmethod
-    def _find_flag(card, flags):
+    def _lookup_label_from_card(card, label_lookup_dict):
```

Comment:
I find 'lookup' a weird word to see in code, compared to 'find', although maybe that's just me.

This still feels like a confusing method to read. Would it be worth refactoring into the Organisation class? Something like `Organisation.find_by_abbreviation()`?